src/axes/timeAxis.ts

```diff
@@ -392,9 +392,17 @@ export module Axes {
       let tickLabelsEnter = tickLabels.enter().append("g").classed(Axis.TICK_LABEL_CLASS, true);
       tickLabelsEnter.append("text");
       let xTranslate = (this._tierLabelPositions[index] === "center" || config.step === 1) ? 0 : this.tickLabelPadding();
-      let yTranslate = this.orientation() === "bottom" ?
-          d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding() :
-          this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding();
+
+      let yTranslate = 0;
+      if (this.orientation() === "bottom") {
+        yTranslate =  d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding();
+      } else {
```

Comment:
Would nested loops be more easier to understand than the 3 branch if statement? Because in the first if statement, we try to pick from `orientation` and the in the nested if statement, we try to pick in `tierlabalpoistion`.